plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1011,7 +1011,8 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
 
     private Optional<Long> getSnapshotId(org.apache.iceberg.Table table, Optional<Long> snapshotId)
     {
-        return snapshotIds.computeIfAbsent(table.toString(), ignored -> snapshotId
+        // table.name() reflects SchemaTableName
```

Comment:
This comment and the commit description is technically correct, but misleading.

We don't care about generic `Table` implementations, since this code is only used with `BaseTable` that we explicitly create in `IcebergUtil.loadIcebergTable()`. We construct the object with our own name using `quotedTableName()` that guarantees proper strings appropriate for a cache key.

The string is not simplify "sufficient" as the cache key: it is a string that we construct explicitly to be the cache key.

Reply:
> We don't care about generic Table implementations, since this code is only used with BaseTable that we create ourselves in IcebergUtil.loadIcebergTable(). We construct the object with our own name using quotedTableName() that guarantees proper strings appropriate for a cache key.

That's what i found out and that;s what the comment was supposed to hint at.
Otherwise, one could suspect the `name()` is just eg an unqualified name , that's not guaranteed to be unique.

Alternative could be to use the `location()` as the cache key. (with a different behavior where to metastore table entries point to same iceberg table).
But, locations are longer than qualified names, so retained name as the key.


Anyway, @electrum  you requested changes on the PR. Help me please understand -- do you want me to change the comment, the cache key, or introduce BaseTable subclass?

Reply:
Since the commit description is misleading, we could drop the description and clarify the code comment:
```java
// table.name() is an encoded version of SchemaTableName
```
Introducing a `BaseTable` subclass would be a better long term improvement.